sycl/source/buffer.cpp

```diff
@@ -121,6 +121,11 @@ void buffer_plain::addOrReplaceAccessorProperties(
 
 size_t buffer_plain::getSize() const { return impl->getSizeInBytes(); }
 
+void buffer_plain::handleRelease(bool DefaultAllocator) const {
+  if (impl.use_count() == 1)
```

Comment:
fixed here https://github.com/intel/llvm/pull/6837/commits/b5e85de484fb026152cfcaa8d3dd8bbc79c158d3

Reply:
Is there a guarantee that neither of the following happens between execution of lines 127 and 128?
- `impl.use_count()` grows to 2 
- `impl.use_count()` drops to 0 
If use, would be good to add a comment.

Reply:
handleRelease is called from buffer destructor. The possible way to increase impl.use_count is creation of buffer copy using operator= or copy constructor which is not expected to be called in parallel with buffer destructor.
use_count to become 0 is also not possible because here we still hold reference.
I will add a comment.